buildcontext/git.go

```diff
@@ -63,7 +63,38 @@ type resolvedGitProject struct {
 	// refs is the git refs
 	refs []string
 	// state is the state holding the git files.
-	state pllb.State
+	state          pllb.State
+	earthfilePaths []string
+}
+
+func (gr *gitResolver) expandWildcard(ctx context.Context, gwClient gwclient.Client, platr *platutil.Resolver, target domain.Target, pattern string) ([]string, error) {
+	if !target.IsRemote() {
+		return nil, errors.Errorf("unexpected local reference %s", target.String())
+	}
+
+	rgp, _, subDir, err := gr.resolveGitProject(ctx, gwClient, platr, target)
+	if err != nil {
+		return nil, err
+	}
+
+	fullPattern := "./" + filepath.Join(subDir, pattern)
```

Comment:
Nit: Not sure if necessary, but should this path be cleaned? Like if it contains things like `././foo/bar/../../buz`?

Reply:
`Join` should do most of the work but `Clean` will strip the `./` prefix. I'll add. 

https://go.dev/play/p/Q2GwzWroqVI

Reply:
Oh yeah, `Join` seems to be doing a clean of its own 👍

Reply:
`Clean` actually strips off the leading `./`, which the following code expects.

Reply:
SG!